test/boost/database_test.cc

```diff
@@ -583,33 +598,54 @@ fs::path table_dir(const replica::column_family& cf) {
     return std::get<data_dictionary::storage_options::local>(cf.get_storage_options().value).dir;
 }
 
-SEASTAR_TEST_CASE(snapshot_works) {
-    return do_with_some_data({"cf"}, [] (cql_test_env& e) {
-        take_snapshot(e).get();
+static auto snapshot_works(const std::string& table_name) {
```

Comment:
Please avoid "auto" return types, it just makes a reader unable to guess what it's supposed to be doing. It's returning `future<>`, right?